lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1092,10 +1092,21 @@ InitKindRequest::evaluate(Evaluator &evaluator, ConstructorDecl *decl) const {
     if (dyn_cast<ClassDecl>(nominal) &&
         !decl->isSynthesized() && isa<ExtensionDecl>(decl->getDeclContext()) &&
         !(decl->getAttrs().hasAttribute<DynamicReplacementAttr>())) {
-      diags.diagnose(decl->getLoc(), diag::designated_init_in_extension,
+          auto classDecl = dyn_cast<ClassDecl>(nominal);
+
+          if (classDecl && 
```

Comment:
`classDecl` will never be null here as its checked earlier by the `dyn_cast`. You can replace the above `dyn_cast` with an `isa` and then the `dyn_cast` here with a `cast`.

Reply:
Yes, you're right. I made the changes that you ask. Thanks!